lib/motions/general-motions.coffee

```diff
@@ -379,11 +379,62 @@ class MoveToMiddleOfScreen extends MoveToScreenLine
     height = lastScreenRow - firstScreenRow
     Math.floor(firstScreenRow + (height / 2))
 
+class ScrollKeepingCursor extends MoveToLine
+  previousFirstScreenRow: 0
+  currentFirstScreenRow: 0
+
+  constructor: (@editor, @vimState) ->
+    super(@editor, @vimState)
+
+  select: (count, options) ->
+    final_destination = @scrollScreen(count)
+    super(count, options)
+    @editor.setScrollTop(final_destination)
+
+  execute: (count) ->
+    final_destination = @scrollScreen(count)
```

Comment:
We usually don't use `snake_case` variable names. I would say call this variable `scrollDestination`.